### PR TITLE
chore: release 3.4.0

### PR DESCRIPTION
**Pull request by SanketD92**
Issue #, if available:
N/A

Description of changes:
Updating changelog for release candidate v3.4.0

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you successfully tested with your changes locally?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,33 @@
...
+* no sagemaker autostop or EC2 stop lag ([#703](https://github.com/awslabs/service-workbench-on-aws/issues/703)) ([8cb199b](https://github.com/awslabs/service-workbench-on-aws/commit/8cb199b8093f5e799d2d87c228930a4929ebebb7))
+* properly handle very long error messages on env update ([#705](https://github.com/awslabs/service-workbench-on-aws/issues/705)) ([d920abd](https://github.com/awslabs/service-workbench-on-aws/commit/d920abd8666eaf905810680aec24428e8ce46124))
+* reset ForceLogout component upon relogin ([#640](https://github.com/awslabs/service-workbench-on-aws/issues/640)) ([5c2aaee](https://github.com/awslabs/service-workbench-on-aws/commit/5c2aaee79428c3d4e2bceb115b77a6eb477a6add))
+* static namespace bug fix ([#615](https://github.com/awslabs/service-workbench-on-aws/issues/615)) ([bacb469](https://github.com/awslabs/service-workbench-on-aws/commit/bacb469d048601cc73f10a3d7145197fbeae8c62))
+* Static namespace fix linted ([#629](https://github.com/awslabs/service-workbench-on-aws/issues/629)) ([f81a04f](https://github.com/awslabs/service-workbench-on-aws/commit/f81a04f777601060c4f741781918374aef0dba25))
```
Comment (maghirardelli) [inline CHANGELOG.md]: Do we always include all fixes? For instance, this is just the linted version of the one above so it doesn't really include changes.

File: main/config/settings/.defaults.yml [hunk 2]
```diff
@@ -172,7 +172,7 @@ auditQldbName: ${self:custom.settings.namespace}-audit
 createServiceCatalogPortfolio: true
 
 # Version number of current release
-versionNumber: '3.3.1'
+versionNumber: '3.4.0'
 
 # Release date of current release
 versionDate: '2021-07-26'
```
Comment (maghirardelli) [inline main/config/settings/.defaults.yml]: Need to change versionDate as well?

File: CHANGELOG.md (hunk 1 above)
Comment (SanketD92) [reply CHANGELOG.md]: I've removed a lot of unnecessary ones. I'll merge the ones you mentioned.

**Review by maghirardelli — APPROVED** [review]
LGTM

**Review by nguyen102 — APPROVED** [review]

**Pull request merged**